scripts/tools/zap/generate.py

```diff
@@ -87,10 +88,27 @@ def runPrettifier(templates_file, output_dir):
     try:
         jsonData = json.loads(Path(templates_file).read_text())
         outputs = [(os.path.join(output_dir, template['output'])) for template in jsonData['templates']]
-        outputs = list(filter(lambda filepath: os.path.splitext(filepath)[1] in listOfSupportedFileExtensions, outputs))
-        os.system('clang-format -i ' + ' '.join(outputs))
+        clangOutputs = list(filter(lambda filepath: os.path.splitext(filepath)[1] in listOfSupportedFileExtensions, outputs))
+        javaOutputs = list(filter(lambda filepath: os.path.splitext(filepath)[1] == ".java", outputs))
+
+        if len(javaOutputs) > 0:
+            # Keep this version in sync with what restyler uses (https://github.com/project-chip/connectedhomeip/blob/master/.restyled.yaml).
+            google_java_format_version = "1.6"
+            google_java_format_url = 'https://github.com/google/google-java-format/releases/download/google-java-format-' + google_java_format_version + '/'
+            google_java_format_jar = 'google-java-format-' + google_java_format_version + '-all-deps.jar'
+            jar_url = google_java_format_url + google_java_format_jar
+            
+            home = str(Path.home())
+            path, http_message = urllib.request.urlretrieve(jar_url, home + '/' + google_java_format_jar)
+            args = ['java', '-jar', path, '--replace']
+            args.extend(javaOutputs)
+            subprocess.check_call(args)
+        if len(clangOutputs) > 0:
+            args = ['clang-format', '-i']
+            args.extend(clangOutputs)
+            subprocess.check_call(args)
```

Comment:
Mind doing 2 separates methods that are called by `runPrettifier` ? Maybe `runClangPrettifier` and `runJavaPrettifier`